src/main/scala/inox/solvers/combinators/NonIncrementalSolver.scala

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2009-2018 EPFL, Lausanne */
+
+package inox
+package solvers
+package combinators
+
+import utils._
+import scala.concurrent.duration._
+import scala.collection.mutable.ArrayBuffer
+
+trait NonIncrementalSolver extends AbstractSolver { self =>
+  import program.trees._
+  import SolverResponses._
+
+  protected def underlying(): AbstractSolver {
+    val program: self.program.type
+    type Trees = self.Trees
+    type Model = self.Model
+  }
+
+  val assertions: IncrementalSet[Trees] = new IncrementalSet[Trees]
```

Comment:
I know I mentioned `IncrementalSet` before, but maybe we should be using `IncrementalSeq` to keep the ordering deterministic (_e.g._ for easier debugging).

Reply:
Fixed, thanks